### PR TITLE
feat(ip): Add completion for monitor subcommand

### PR DESCRIPTION
**Pull request by yedayak**
Currently it stops after completing the first option, although you should be able to choose multiple options.
I'll be happy for some pointers on how I can make it keep completing after the first option, but I think this is enough on its own.

### CODE REVIEW

File: completions/ip [hunk 1]
```diff
@@ -357,7 +357,14 @@ _comp_cmd_ip()
                 all) ;;
                 *)
                     ((cword == subcword)) &&
-                        _comp_compgen -- -W 'help all'
+                        _comp_compgen_split -- "help all $(
+                            {
+                                $1 -c=never monitor help || $1 monitor help
+                            } 2>&1 | command sed -e \
+                                '/OBJECTS := /,/|.*$/!d' -e \
```
Comment (akinomyoga) [inline completions/ip:364]: I think `/|.*$/` matches any line containing `|`, so this condition fails when the number of lines of OBJECTS becomes three or more. For example, with this mock text:

```bash
$ printf '%s\n' 'OBJECTS := address | link |' '    mroute | neigh |' '    nexthop | stats'
OBJECTS := address | link |
    mroute | neigh |
    nexthop | stats
```

The range only selects the first two:

```bash
$ printf '%s\n' 'OBJECTS := address | link |' '    mroute | neigh |' '    nexthop | stats' | sed -ne '/OBJECTS := /,/|.*$/p'
OBJECTS := address | link |
    mroute | neigh |
$
```

File: completions/ip [hunk 2]
```diff
@@ -357,7 +357,14 @@ _comp_cmd_ip()
                 all) ;;
                 *)
                     ((cword == subcword)) &&
-                        _comp_compgen -- -W 'help all'
+                        _comp_compgen_split -- "help all $(
+                            {
+                                $1 -c=never monitor help || $1 monitor help
```
Comment (akinomyoga) [inline completions/ip:362]:
```suggestion
                                "$1" -c=never monitor help || "$1" monitor help
```

It's better to quote them, though there are still many unquoted cases in the codebase.

**Review by akinomyoga — CHANGES_REQUESTED** [review]
Thank you for your contribution. The change in this PR is generally nice, but I think some minor points should be updated.

File: completions/ip (hunk 2 above)
Comment (yedayak) [reply completions/ip]: Fixed

File: completions/ip (hunk 1 above)
Comment (yedayak) [reply completions/ip]: Nicely spotted, I changed it to match until a line that don't end with with `|`.

Comment (akinomyoga) [reply completions/ip]: Thanks!

**Review by akinomyoga — APPROVED** [review]

**Comment by scop** [comment]
Thanks!

**Pull request merged**